### ISSUE TITLE
Vuex TypeScript documentation seems incomplete

### ISSUE THREAD

**Issue by francoism90** [quasarframework/quasar#8500]
I'm new to TypeScript, however there seem to be a few things missing the Quasar V2 when browsing the [PreFretch feature](https://next.quasar.dev/quasar-cli/prefetch-feature#Usage-with-TypeScript) page. According to the documentation hinting should work when adding the store state.

Take the following example (this includes some official [Vuex documentation](https://next.vuex.vuejs.org/guide/typescript-support.html#simplifying-usestore-usage) suggestions):
```
// interfaces/store.ts
export interface SessionState {
  timestamp: Date | null;
  token: string | null;
}

export interface StoreState {
  session: SessionState;
}

// store/index.ts
import { store } from 'quasar/wrappers'
import { InjectionKey } from 'vue'
import { createStore, useStore as baseUseStore, Store } from 'vuex'
import { StoreState } from 'src/interfaces/store'
import session from './session'

export const key: InjectionKey<Store<StoreState>> = Symbol('vuex_key')

export default store(function (/* { ssrContext } */) {
  const Store = createStore<StoreState>({
    modules: {
      session
    },
    strict: !!process.env.DEBUGGING
  })

  return Store
})

export function useStore () {
  return baseUseStore(key)
}

// Index.vue
<script lang="ts">
import { preFetch } from 'quasar/wrappers'
import { Store } from 'vuex'
import { defineComponent } from 'vue'
import { StoreState } from 'src/interfaces/store'

export default defineComponent({
  name: 'PageIndex',

  preFetch: preFetch<Store<StoreState>>(({ store }) => {
    console.log(store.state.session.timestamp) // Property 'session' does not exist on type 'Store<StoreState>'.
  })
})
</script>
```

The same doesn't work with the [Store Code Splitting](https://next.quasar.dev/quasar-cli/prefetch-feature#Store-Code-Splitting) example, you need to call the correct store:
```
<template>
  <q-page class="container">

  </q-page>
</template>

<script lang="ts">
import { useStore } from 'src/store' // instead of import { useStore } from 'vuex'
import { defineComponent } from 'vue'

export default defineComponent({
  name: 'PageIndex',

  setup () {
    const $store = useStore()

    console.log($store.state.session.timestamp)
  }
})
</script>
```

Maybe I'm wrong, sorry about that. I don't think this is a support question as the example given in the documentation doesn't provide typehinting.

If any more information is needed, please let me know.

**Comment by francoism90** [comment]
This offers completion and prevents TS errors, however maybe Quasar/Vuex should provide a more friendly solution:

```
// src/types/vuex.d.ts -> offers $store completion
import { StoreState } from 'src/interfaces/store'
import { Store } from 'vuex'

declare module '@vue/runtime-core' {
  interface ComponentCustomProperties {
    $store: Store<StoreState>
  }
}

// src/store/index.ts
import { store } from 'quasar/wrappers'
import { createStore } from 'vuex'
import { StoreState } from 'src/interfaces/store'
import session from './session'

export default store(function (/* { ssrContext } */) {
  const Store = createStore<StoreState>({
    modules: {
      session
    },
    strict: !!process.env.DEBUGGING
  })

  return Store
})

// src/boot/store.ts
import { boot } from 'quasar/wrappers'
import { key } from 'src/services/store'

export default boot(({ app, store }) => {
  app.provide(key, store)
})

// src/services/store
import { StoreState } from 'src/interfaces/store'
import { InjectionKey } from 'vue'
import { useStore as baseUseStore, Store } from 'vuex'

export const key: InjectionKey<Store<StoreState>> = Symbol('vuex')

export function useStore () {
  return baseUseStore(key)
}

// src/pages/Index.vue
<template>
  <q-page class="container" />
</template>

<script lang="ts">
import { defineComponent } from 'vue'
import { useStore } from 'src/services/store'

export default defineComponent({
  name: 'PageIndex',

  setup () {
    const store = useStore()

    console.log(store.state.session.timestamp)

    const test = store.dispatch('foo/bar')

    console.log(test)
  }

  computed: {
    foo () {
      return this.$store.state.session.bar
    },
  }
})
</script>
```

Maybe Quasar should provide an `InjectionKey` by default for Vuex and/or offer some hooks?

**Comment by IlCallo** [comment]
> According to the documentation hinting should work when adding the store state.

Our fault, we simplified the parameter, you only need to provide `StateInterface` instead of `Store<StateInterface>` since v2, gonna fix it now.

#8511 should fix other mentioned issues as you pointed out on the PR

**Comment by rstoenescu** [comment]
Enhancement will be available through q/app v3.0.0-beta.11
